scripts/template_matcher.py

```diff
@@ -58,16 +72,65 @@ def _compute_prediction(self, k, img, kp, des):
         k: template image for comparison, img: scene image
         kp: keypoints from scene image,   des: descriptors from scene image
         """
-
-        # TODO: find corresponding points in the input image and the templae image
-        #       put keypoints from template image in template_pts
-        #       put corresponding keypoints from input image in img_pts
-
-
-        #TODO: change img to img_T once you do the homography transform
-        visual_diff = compare_images(img, self.signs[k])
+        #For every key point on template, compare it to all input key points and pick the closest to match with
+
+
+        template_pts = []
+        img_pts = []
+
+        for temp in self.kps[k]:
+            #Improbably high distance
+            lowest_dist = 1000000000
+            curr_temp = None
+            curr_scene = None
+            #Calculate the distance between keypoints and take the best for each template keypoint
+            for keyp in kp:
```

Comment:
Given a good bounding box, this should work pretty well, but a better way to match keypoints is to compare the area around the keypoint. Each keypoint has a descriptor associated with it which is a 128 element vector representing the image around the keypoint. If you do basically the the same thing to find the closest keypoint in those 128 dimensions, then you have a match which will hold regardless of translations rotations or scaling within the image